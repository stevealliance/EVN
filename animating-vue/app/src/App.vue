<template>
  <div id="app">
    <div id="nav">
      <router-link to="/">Modal</router-link> |
      <router-link to="/list">List</router-link>
    </div>
    <transition name="slide-fade" mode="out-in">
      <router-view />
    </transition>
  </div>
</template>

<style>
#app {
  font-family: Avenir, Helvetica, Arial, sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  text-align: center;
  color: #2c3e50;
  display: flex;
  justify-content: center;
  flex-wrap: wrap;
  flex-direction: column;
  align-items: center;
}

.home {
  width: 100%;
  display: flex;
  justify-content: center;
  flex-wrap: wrap;
  flex-direction: column;
  align-items: center;
}

.home button, .list button {
  border:0px;
  padding: 12px;
  background-image: linear-gradient(to right, #FF5F6D, #FFC371);
  color: #fff;
  width: 80px;
  border-radius: 4px;
}

.modal {
  width: 50%;
  padding-bottom: 200px;
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: #ddd;
  border: 1px solid #ccc;
  border-radius: 6px;
  box-shadow: 0px 3px 8px #383838ab
}

.modal button {
  margin-top: 20px;
}

#nav {
  padding: 30px;
}

#nav a {
  font-weight: bold;
  color: #FFC371;
}

#nav a.router-link-exact-active {
  color: #FF5F6D;
}

.list {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.list li {
  list-style-type: none;
}

.input-container {
  display: inline-flex;
}

.input-container input {
  margin-right:5px;
  padding: 5px;
  border-radius: 5px;
  border: 2px solid;
  border-image-source: linear-gradient(to right, #FF5F6D, #FFC371);
  border-image-slice: 1;
}
/***** TRANSITIONS ******/

.fade-enter {
  opacity: 0;

}

.fade-enter-active, .fade-leave-active {
  transition: opacity 0.5s ease-out;
}

.fade-leave-to {
  opacity: 0;
}


.slide-fade-enter {
  transform: translateX(10px);
  opacity: 0;

}

.slide-fade-enter-active, .slide-fade-leave-active {
  transition: all 0.3s ease;
}

.slide-fade-leave-to {
  transform: translateX(-10px);
  opacity: 0;
}

.slide-up-enter {
  transform: translateY(10px);
  opacity: 0;
}

.slide-up-enter-active {
  transition: all 0.3s ease;
}

</style>
