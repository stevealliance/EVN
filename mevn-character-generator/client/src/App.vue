<template>
  <div id="app">
    <div id="nav">
      <router-link to="/">View</router-link> |
      <router-link to="/create">Create</router-link>
    </div>
    <transition name="slide-fade" mode="out-in">
      <router-view />
    </transition>
  </div>
</template>

<style>
body {
  background-color:#73937E;
}
#app {
  font-family: Avenir, Helvetica, Arial, sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  text-align: center;
  color: #2c3e50;
}

#nav {
  padding: 30px;
}

#nav a {
  font-weight: bold;
  color: #2c3e50;
}

#nav a.router-link-exact-active {
  color: #42b983;
}

.slide-fade-enter {
  transform: translateX(10px);
  opacity: 0;

}

.slide-fade-enter-active, .slide-fade-leave-active {
  transition: all .3s ease;
}

.slide-fade-leave-to {
  transform: translateX(-10px);
  opacity: 0;
}

.slide-up-enter {
  transform: translateY(10px);
  opacity: 0;
}

.slide-up-enter-active {
  transition: all .4s ease;
}
</style>
