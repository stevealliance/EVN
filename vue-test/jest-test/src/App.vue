<template>
  <div id="app">
    <MessageList>
      <Message
        @message-clicked="handleMessageClick"
        :message="message"
        v-for="message in messages"
        :key="message"/>
    </MessageList>
  </div>
</template>

<script>
import MessageList from './components/MessageList'
import Message from "./components/Message";

export default {
  name: "app",
  data: () => ({ messages: ["Hey John", "Howdy Paco"] }),
  methods: {
    handleMessageClick(message) {
      console.log(message);
    }
  },
  components: {
    MessageList,
    Message
  }
};
</script>

<style>
#app {
  font-family: 'Avenir', Helvetica, Arial, sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  text-align: center;
  color: #2c3e50;
  margin-top: 60px;
}
</style>
