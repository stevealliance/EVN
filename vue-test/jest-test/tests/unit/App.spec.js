import App from '@/App'
import { shallowMount } from '@vue/test-utils'

describe('App.test.js',() => {
    let component
    
    beforeEach(() => {
        component = shallowMount(App)
        component.setData({ messages: ['Dog', 'Cat', 'Fish']});
    })

    it('should equals to ["Dog", "Cat", "Fish"]', () => {
        expect(component.vm.messages).toEqual(['Dog', 'Cat', 'Fish'])
    })

    it('has the expected html structure', () => {
        expect(component.element).toMatchSnapshot()
    })
})