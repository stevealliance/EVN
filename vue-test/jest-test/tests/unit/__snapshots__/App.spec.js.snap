// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.test.js has the expected html structure 1`] = `
<div
  id="app"
>
  <messagelist-stub>
    <header>
      
      Awesome header
    
    </header>
     
    <message-stub
      author="Dylan"
      message="Dog"
    />
    <message-stub
      author="Dylan"
      message="Cat"
    />
    <message-stub
      author="Dylan"
      message="Fish"
    />
  </messagelist-stub>
</div>
`;
